### PR TITLE
Add sap fieldglass approval connector

### PR DESCRIPTION
**Pull request by SanduDS**
# Description

Generate the SAP Fieldglass Approval connector using the official OpenAPI specification.
https://api.sap.com/api/approvals/resource

Fixes https://github.com/ballerina-platform/ballerina-extended-library/issues/275

One line release note: 
- Generate the SAP Fieldglass Approval connector using the official OpenAPI specification.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Not tested

**Test Configuration**:
* Ballerina Version: SL Beta 6
* Operating System: Ubuntu 20.04
* Java SDK: 11

# Checklist:

### Changes to OpenAPI definition

- [x] Modified `info` section of the OpenAPI definition file - [Example](https://github.com/ballerina-platform/ballerina-extended-library/discussions/74)

    - `description` - what the connector is about 
    - `x-ballerina-init-description` OpenAPI extension - connector initialization details 
    - `x-ballerina-display` OpenAPI extension - connector name and path to the connector icon 

- [x] Added proper description each API Key, if API Key authentication is defined in OpenAPI definition file 

### Changes to connector module

- [x] Added license header at the top of each .bal file. 
- [x] Added Package.md as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/77)
- [x] Added Module.md for the module of the  connector as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/78)
- [x] Added connector icon to the root of the connector. Icon needs to be a `png` of 200x200 px size with name `icon.png`
- [x] Added Ballerina.toml file with following information. For keywords refer to [guide](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72)

### Verifying connector module

- [x] Compiled the connector successfully with the targeted ballerina version. 
- [ ] Conducted smoke tests on the connector (Recommended when OpenAPI definition is obtain from an unofficial source)

### Security checks
 - [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: openapi/sap.fieldglass.approval/client.bal [hunk 1]
```diff
@@ -0,0 +1,134 @@
+import ballerina/http;
```
Comment (aneeshafedo) [inline openapi/sap.fieldglass.approval/client.bal:1]: Missing license header

File: openapi/sap.fieldglass.approval/types.bal [hunk 2]
```diff
@@ -0,0 +1,77 @@
+public type Status record {
```
Comment (aneeshafedo) [inline openapi/sap.fieldglass.approval/types.bal:1]: Missing license header

File: openapi/sap.fieldglass.approval/utils.bal [hunk 3]
```diff
@@ -0,0 +1,223 @@
+import ballerina/url;
```
Comment (aneeshafedo) [inline openapi/sap.fieldglass.approval/utils.bal:1]: Missing license header

**Review by LakshanSS — APPROVED** [review]

**Comment by LakshanSS** [comment]
Need to add icon

**Comment by LakshanSS** [comment]
> Need to add icon

Added

**Pull request merged**